### Julia Project  
#### Given two points as the endpoints of a diameter of a circle: $(x_1,y_1)$ and $(x_2, y_2)$, create Julia function(s) to do the following: 
1) Find the center of the circle (equivalent to midpoint): $( \frac{x_1+x_2}{2}, \frac{y_1+y_2}{2} )$ where $h = \frac{x_1+x_2}{2}$ and $k = \frac{y_1+y_2}{2}$ <br>
2) Find the radius of the circle (use Pythagorean Theorem): $r^2 = (y_1-y_2)^2 + (x_1-x_2)^2$ <br>
3) Find a circle equation: $(x-h)^2 + (y-k)^2 = r^2$ <br>
4) Find the slope of the given diameter: $m = \frac{x_1 - x_2}{y_1 - y_2}$ <br>
5) Find the equation of the perpendicular bisector of the given diameter (hint: perpendicular means its slope is $\frac{-1}{m}$):<br>
$$(y - y_1) = \frac{-1}{m}(x - x_1)$$ 
6) Find end points of the second diameter <br>



#### Answer to #1 problem
Insert the given two points, $(x_1, y_1)$ and $(x_2, y_2)$, as inputs for the function "find_center". (Note: $h = \frac{x_1+x_2}{2}$ and $k = \frac{y_1+y_2}{2}$) <br>

In [1]:
#Insert given two points, (x1, y1) and (x2, y2), as inputs for the function "find_center"
# Answer to #1 question

function find_center(x1,y1,x2,y2)
    h = (x1+x2)/2                # h is x-center
    k = (y1+y2)/2                # k is y-center
    return((h,k))
end

find_center (generic function with 1 method)

In [2]:
#Test with two given two points, (x1,y1)=(5,3) and (x2,y2) = (-1, -5)

find_center(5,3,-1,-5)

(2.0, -1.0)

#### Answer to #2 problem
Insert the given two points, $(x_1, y_1)$ and $(x_2, y_2)$, as inputs for the function "find_radius".<br>
(Note: $h = \frac{x_1+x_2}{2}$ and $k = \frac{y_1+y_2}{2}$) <br>

In [3]:
#Insert given two points, (x1, y1) and (x2, y2), as inputs for the function "find_radius"
# Answer to #2 question

function find_radius(x1,y1,x2,y2)
    h = (x1+x2)/2                      # h is x-center
    k = (y1+y2)/2                      # k is y-center
    r = sqrt( (x1-h)^2 + (y1-k)^2 )    # use Pythagorean Theorem to find r^2
    return(r)
end

find_radius (generic function with 1 method)

In [4]:
#Test with two given two points, (x1,y1)=(5,3) and (x2,y2) = (-1, -5)

find_radius(5,3,-1,-5)

5.0

#### Answer to #3 problem
Problem 3 does not require to create a Julia function.  Just use problem 1 and 2 to do the problem 3

$(x-h)^2 + (y-k)^2 = r^2$ where you get $h$ and $k$ from "find_center" and $r$ from "find_radius" functions.

For the given two end points its circle equation is $(x - 2)^2 + (y + 1)^2 = 25$ where $h = 2, k = -1$, and $r = 5$

#### Answer to #4 problem
Insert the given two points, $(x_1, y_1)$ and $(x_2, y_2)$, as inputs for the function "find_slope"

In [5]:
## Answer to #4 problem
# Insert the given two points, (x1, y1) and (x2, y2), as inputs for the function "find_slope"

function find_slope(x1,y1,x2,y2)
    m = (y1-y2)/(x1-x2)                 # m = slope
    return(m)
end

find_slope (generic function with 1 method)

In [6]:
#Test with two given two points, (x1,y1)=(5,3) and (x2,y2) = (-1, -5)

find_slope(5,3,-1,-5)

1.3333333333333333

#### Answer to #5 problem
Insert the given two points, $(x_1, y_1)$ and $(x_2, y_2)$, as inputs for the function "find_perd_yint"

After find the y-intercept and using problems 4 we can find an equation $y = \frac{-1}{m}\cdot x + b$


In [7]:
## Answer to #5 problem
# Insert the given two points, (x1, y1) and (x2, y2), as inputs for the function "find_yinterceipt"

function find_perd_yint(x1,y1,x2,y2)
    m = (y1-y2)/(x1-x2)                 # m = slope
    perd_m = -1/m                       # perpendicular slope means -1/m
    b = y1 - perd_m*x1                  # find b = y-intercept
    return(b)                           # return y-intercetp 
end

find_perd_yint (generic function with 1 method)

In [8]:
#Test with two given two points, (x1,y1)=(5,3) and (x2,y2) = (-1, -5)

find_perd_yint(5,3,-1,-5)

6.75

#### For the given two end points, its perpendicular equation is $y = 1.33x + 6.75$ where $\frac{-1}{m} = \frac{-1}{1.33} = -0.75$ and $b = 6.75$

#### Answer to #6 problem
Insert the given two points, $(x_1, y_1)$ and $(x_2, y_2)$, as inputs for the function "find_two_endpoints"
In order to find two end poitns of the other diameter we need to know $height=|h-y_1|$ and $width = |k-x_1|$. <br>
New points are: <br>
$x_3 = x_{center}$(or $h$) + width, <br>
$y_3 = y_{center}$(or $k$) - height, <br>
$x_4 = x_{center}$(or $h$) - width, and <br>
$y_4 = y_{center}$(or $k$) + height    

In [9]:
#Insert given two points, (x1, y1) and (x2, y2), as inputs for the function "find_two_endpoints"
# Answer to #6 question

function find_two_endpoints(x1,y1,x2,y2)
    h = (x1+x2)/2                          # h = x_center 
    k = (y1+y2)/2                          # k = y_center
    height = abs(k-y2)
    width = abs(h-x2)    
    x1_endpoint = h + width         # find the first x end point
    x2_endpoint = h - width         # find the second x end point
    y1_endpoint = k - height        # find the first y end point
    y2_endpoint = k + height        # find the second y end point
    return((x1_endpoint,y1_endpoint), (x2_endpoint,y2_endpoint))   # return two endpoints
end

find_two_endpoints (generic function with 1 method)

In [10]:
find_two_endpoints(5,3,-1,-5)

((5.0, -5.0), (-1.0, 3.0))